Network/Extensions/ThreadExtensions.cs

```diff
@@ -27,22 +28,33 @@
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // ***********************************************************************
+
 #endregion Licence - LGPLv3
+
 using System.Threading;
 
 namespace Network.Extensions
 {
     /// <summary>
-    /// Offers some nice extensions for a thread instance.
+    /// Provides additional functionality to the <see cref="Thread"/>
+    /// class, that isn't available under .NET Standard 2.0.
     /// </summary>
     internal static class ThreadExtensions
     {
+        #region Methods
+
         /// <summary>
-        /// Aborts a thread and catches all the exceptions if some occurs.
+        /// Allows for a <see cref="Thread"/> object to be aborted in a program
+        /// running under the .NET Standard C# implementation.
         /// </summary>
-        /// <param name="thread">The thread to abort.</param>
-        /// <returns>If an exception occured.</returns>
-        internal static bool AbortSave(this Thread thread)
+        /// <param name="thread">
+        /// The <see cref="Thread"/> instance this extension method affects.
+        /// </param>
+        /// <returns>
+        /// Whether the <see cref="Thread.Abort(object)"/> method raised an
+        /// exception.
+        /// </returns>
+        public static bool AbortSave(this Thread thread)
```

Comment:
why public? Try to keep as much internal/private as possible. Devs shouldn't read and care about stuff they are never going to use. Basically, every internal and private u made public, rethink what the benefit would be. If there is no benefit, turn it back to private/internal

Reply:
Since the method is in an internal class, it doesnt matter if it is marked public. It still wont be able to be read outside fo the class, as the class cannot be accessed. In any such cases (public method in internal class) the lowest access modifier is used; public methods in internal classes function EXACTLY like internal methods. Public is shorter to write, so i use it.

Reply:
Ok, didn't know that. However, for me, I like consistency and it confuses me to see a public method in an internal class. I prefer internal in an internal class and public in public ;)

Reply:
However, leave it like this for the moment. Need to think about it.